src/libcore/future/pending.rs

```diff
@@ -0,0 +1,61 @@
+use crate::future::Future;
+use crate::marker;
+use crate::pin::Pin;
+use crate::task::{Context, Poll};
+
+/// Creates a future which never resolves, representing a computation that never
+/// finishes.
+///
+/// This `struct` is created by the [`pending`] function. See its
+/// documentation for more.
+///
+/// [`pending`]: fn.pending.html
+#[unstable(feature = "future_readiness_fns", issue = "none")]
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct Pending<T> {
+    _data: marker::PhantomData<T>,
```

Comment:
I would expect a `PhantomData<fn() -> T>` here, as we only produce T, without storing it, so we should bypass dropck and be covariant.

Reply:
@matklad oh nice, I didn't know that made a difference. Updated with your suggestion!

Reply:
I've tried to come up with example, and either me or rustc is buggy: https://github.com/rust-lang/rust/issues/70841